legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/Planner.java

```diff
@@ -257,8 +252,30 @@ protected PlannerOptions options()
 
     public LogicalPlan buildLogicalPlanForMetadataIngest(Resources resources)
     {
-        List<DatasetFilter> stagingFilters = LogicalPlanUtils.getDatasetFilters(originalStagingDataset());
-        return LogicalPlan.of(Arrays.asList(metadataUtils.insertMetaData(mainTableName, batchStartTimestamp, batchEndTimestamp, stagingFilters)));
+        // Save staging filters into batch_source_info column
+        Map<String, Object> batchSourceInfoMap = new HashMap<>();
+        if (originalStagingDataset() instanceof DerivedDataset)
+        {
+            DerivedDataset derivedDataset = (DerivedDataset) originalStagingDataset();
+            LogicalPlanUtils.jsonifyStagingFilters(batchSourceInfoMap, derivedDataset.datasetFilters());
```

Comment:
We can let the jsonifyStagingFilters return the json as a map

Reply:
Sure, changed